File: WordBattle/WordBattle/mobile/lib/utilize/letter_pool.dart
import 'dart:math';

class LetterPool {
  static final List<String> _pool = _createPool();

  static List<String> _createPool() {
    List<String> letters = [];

    void addLetters(String letter, int count) {
      for (int i = 0; i < count; i++) {
        letters.add(letter);
      }
    }

    addLetters('A', 12);
    addLetters('B', 2);
    addLetters('C', 2);
    addLetters('Ç', 2);
    addLetters('D', 3);
    addLetters('E', 8);
    addLetters('F', 2);
    addLetters('G', 3);
    addLetters('Ğ', 1);
    addLetters('H', 1);
    addLetters('I', 6);
    addLetters('İ', 4);
    addLetters('J', 1);
    addLetters('K', 7);
    addLetters('L', 5);
    addLetters('M', 4);
    addLetters('N', 7);
    addLetters('O', 3);
    addLetters('Ö', 1);
    addLetters('P', 2);
    addLetters('R', 6);
    addLetters('S', 5);
    addLetters('Ş', 2);
    addLetters('T', 5);
    addLetters('U', 3);
    addLetters('Ü', 2);
    addLetters('V', 2);
    addLetters('Y', 3);
    addLetters('Z', 2);

    return letters;
  }

  static List<String> drawLetters(int count) {
    final random = Random();
    List<String> drawn = [];

    for (int i = 0; i < count; i++) {
      if (_pool.isEmpty) break;
      int index = random.nextInt(_pool.length);
      drawn.add(_pool.removeAt(index)); // 🔥 Direkt _pool'dan çekiyoruz
    }

    return drawn;
  }

  static int get remainingLetters => _pool.length;
}

File: WordBattle/WordBattle/backend/app/letter_pool.py
import random

class LetterPool:
    def __init__(self):
        self._pool = self._create_pool()

    def _create_pool(self):
        letters = []

        def add_letters(letter, count):
            for _ in range(count):
                letters.append(letter)

        add_letters('A', 12)
        add_letters('B', 2)
        add_letters('C', 2)
        add_letters('Ç', 2)
        add_letters('D', 2)
        add_letters('E', 8)
        add_letters('F', 1)
        add_letters('G', 1)
        add_letters('Ğ', 1)
        add_letters('H', 1)
        add_letters('I', 4)
        add_letters('İ', 7)
        add_letters('J', 1)
        add_letters('K', 7)
        add_letters('L', 7)
        add_letters('M', 4)
        add_letters('N', 5)
        add_letters('O', 3)
        add_letters('Ö', 1)
        add_letters('P', 1)
        add_letters('R', 6)
        add_letters('S', 3)
        add_letters('Ş', 2)
        add_letters('T', 5)
        add_letters('U', 3)
        add_letters('Ü', 2)
        add_letters('V', 1)
        add_letters('Y', 2)
        add_letters('Z', 2)

        return letters

    def draw_letters(self, count):
        drawn = []
        for _ in range(count):
            if not self._pool:
                break
            index = random.randint(0, len(self._pool) - 1)
            drawn_letter = self._pool.pop(index)
            drawn.append(drawn_letter)
        print(f"🎯 draw_letters() fonksiyonu: Çekilen harfler: {drawn}")
        return drawn

    def remaining_letters(self):
        return len(self._pool)

    def get_pool(self):
        return list(self._pool)
